src/test/java/picard/analysis/CollectRnaSeqMetricsTest.java

```diff
@@ -382,6 +382,90 @@ public void testTranscriptionStrandMetrics() throws Exception {
         Assert.assertEquals(metrics.PCT_R2_TRANSCRIPT_STRAND_READS, 0.333333);
     }
 
+    @DataProvider(name = "endBiasValues")
+    public static Object[][] endBiasValues() {
+        return new Object[][] {
+                // Case 1 - 25 nt in, ends should both be 0 coverage.
+                {
+                    25,
+                    0.0,
+                    0.0,
+                    0.0
+                },
+                // Case 2 - 50 nt in, 5' should be 0 coverage and 3' should be partially covered.
+                {
+                    50,
+                    1.543624,
+                    0.0,
+                    0.0
+                },
+                // Case 3 - 100 nt in, 3' and 5' should both have coverage.
+                {
+                    100,
+                    2.449664,
+                    1.677852,
+                    0.684932
+                }
+        };
+    }
+
+    @Test(dataProvider = "endBiasValues")
+    public void testBiasEndBiasAdjust(int end_bias_size, double expected_3prime, double expected_5prime, double expected_ratio) throws Exception {
+        final String sequence = "chr1";
+
+        // Create some alignments that hit the ribosomal sequence, various parts of the gene, and intergenic.
+        final SAMRecordSetBuilder builder = new SAMRecordSetBuilder(true, SAMFileHeader.SortOrder.coordinate);
+
+        // Set seed so that strandedness is consistent among runs.
+        builder.setRandomSeed(0);
+        final int sequenceIndex = builder.getHeader().getSequenceIndex(sequence);
+
+        final int read_length = 150;
+        final int n_fragments = 30;
+        final String read_cigar = read_length + "M";
+        final int five_prime_start = 51;
+        final int three_prime_start = 975 - read_length;
+        final String refFlatStart = "0";
+        final String refFlatEnd = "1000";
```

Comment:
I think it could be a little clearer where these numbers are coming from with a small adjustment.

```suggestion
        final String txStart = 0;
        final String txEnd = 1000;
        final int five_prime_start = txStart + 51;
        final int three_prime_start = txEnd - 25 - read_length;
        final String refFlatStart = txStart.toString();
        final String refFlatEnd = tsEnd.toString();

```